documentation updates (#951)
elastic/apm-agent-go: docs/instrumenting.asciidoc

Package apmhttp also provides functions for instrumenting an `http.Client` or `http.RoundTripper`
such that outgoing requests are traced as spans, if the request context includes a transaction.
When performing the request, the enclosing context should be propagated by using
https://golang.org/pkg/net/http/#Request.WithContext[http.Request.WithContext], or a helper, such as those provided by https://golang.org/x/net/context/ctxhttp.

> Client spans are not ended until the response body is fully consumed or closed.
> Failure to do either of these will result in the span not being sent.

Reviewer: ```suggestion
Client spans are not ended until the response body is fully consumed or closed.
If you fail to do either, the span will not be sent.
```

Reviewer: I think we probably want to address the user here, especially if they have to take action. Plus...it takes fewer words.
